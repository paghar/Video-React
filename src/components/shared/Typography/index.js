//@flow
import { Wrapper } from './index.style';
import React from 'react';
type Font = {
  fontSize: Object | string,
  fontName: Object | string,
  fontWeight: Object | string,
  lineHeight: Object | string,
};
type Props = {
  font: Font,
  color?: Object | string,
  hoverable?: boolean,
  children: string,
  className?: string,
};
const Typography = ({
  font: { fontName, fontSize, fontWeight, lineHeight },
  color,
  hoverable,
  children,
  className,
}: Props) => {
  return (
    <Wrapper
      fontName={fontName}
      sizing={fontSize}
      weight={fontWeight}
      lineHeight={lineHeight}
      textColor={color}
      hoverable={hoverable}
      className={className}
    >
      {children}
    </Wrapper>
  );
};

export default Typography;
