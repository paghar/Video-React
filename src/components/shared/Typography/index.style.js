//@flow
import styled from 'styled-components';

export const Wrapper = styled.span`
  font-family: ${({ fontName }) => fontName};
  font-size: ${({ sizing }) => sizing};
  font-weight: ${({ weight }) => weight};
  line-height: ${({ lineHeight }) => lineHeight};
  color: ${({ textColor }) => textColor};
  pointer-events: ${({ hoverable }) => (hoverable ? 'none' : 'auto')};
  ${({ hoverable }) =>
    hoverable
      ? `
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  user-select: none;`
      : ''}
`;
