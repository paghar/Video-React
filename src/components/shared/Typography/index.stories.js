//@flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  Title1,
  Display1,
  Display2,
  Display3,
  Display4,
  Display5,
  Display6,
} from 'theme/fonts';
import Typography from '.';

storiesOf('Typography', module)
  .add('Title1', () => (
    <Typography font={Title1} hoverable color="red">
      Title 1
    </Typography>
  ))
  .add('Display1', () => (
    <Typography font={Display1} color="red">
      Display 1
    </Typography>
  ))
  .add('Display2', () => (
    <Typography font={Display2} color="green">
      Display 2
    </Typography>
  ))
  .add('Display3', () => (
    <Typography font={Display3} color="black">
      Display 3
    </Typography>
  ))
  .add('Display4', () => (
    <Typography font={Display4} color="red">
      Display 4
    </Typography>
  ))
  .add('Display5', () => (
    <Typography font={Display5} color="red">
      Display 5
    </Typography>
  ))
  .add('Display6', () => (
    <Typography font={Display6} color="red">
      Display 6
    </Typography>
  ));
