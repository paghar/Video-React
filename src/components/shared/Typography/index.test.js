//@flow
import React from 'react';
import Typography from '.';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('<Typography />', () => {
  it('checks the text to be exist', () => {
    const wrapper = shallow(
      <Typography
        color="red"
        font={{
          fontSize: '20px',
          fontWeight: '500',
          lineHeight: '60px',
          fontName: "'Red Hat Text', sans-serif",
        }}
        hoverable
      >
        hello
      </Typography>
    );
    expect(wrapper.text()).toBe('hello');
  });
  it('checks the UI', () => {
    const wrapper = renderer
      .create(
        <Typography
          hoverable
          font={{
            fontSize: '10px',
            fontWeight: '400',
            lineHeight: '60px',
            fontName: "'Red Hat Text', sans-serif",
          }}
          color="red"
        >
          hii
        </Typography>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('has checks the UI', () => {
    const wrapper = renderer
      .create(
        <Typography
          font={{
            fontSize: '15px',
            fontWeight: '600',
            lineHeight: '50px',
            fontName: "'Red Hat Text', sans-serif",
          }}
          color="red"
        >
          test
        </Typography>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
