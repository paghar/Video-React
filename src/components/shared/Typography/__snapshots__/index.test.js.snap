// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> checks the UI 1`] = `
.c0 {
  font-family: 'Red Hat Text',sans-serif;
  font-size: 10px;
  font-weight: 400;
  line-height: 60px;
  color: red;
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<span
  className="c0"
>
  hii
</span>
`;

exports[`<Typography /> has checks the UI 1`] = `
.c0 {
  font-family: 'Red Hat Text',sans-serif;
  font-size: 15px;
  font-weight: 600;
  line-height: 50px;
  color: red;
  pointer-events: auto;
}

<span
  className="c0"
>
  test
</span>
`;
